lib/s3_file_uploader.rb

```diff
@@ -0,0 +1,18 @@
+class S3FileUploader
+  def self.save_file_to_s3(filename, csv)
+    connection = Fog::Storage.new(
+      provider: "AWS",
+      region: ENV["AWS_REGION"],
+      aws_access_key_id: ENV["AWS_ACCESS_KEY_ID"],
+      aws_secret_access_key: ENV["AWS_SECRET_ACCESS_KEY"],
+    )
+
+    directory = connection.directories.get(ENV["AWS_S3_BUCKET_NAME"])
+
+    directory.files.create( # rubocop:disable Rails/SaveBang
```

Comment:
Can you explain in the commit why this is disabled? I can't see this being used in the two PRs you linked to.
And would it be possible to improve the code to fix the linting error rather than ignore it?

Reply:
The problem is that `rubocop-rails` is full of false positives: it assumes that a `create` method is always on an ActiveRecord object and has an associated `create!` method, and that the latter should be used if the return value isn't needed. The problem is that it has no way of telling when the object in question is actually an ActiveRecord model.

This particular one seems to get disabled a lot in whitehall:

```
$ rg 'rubocop:disable Rails/SaveBang' | wc -l
      48
$ rg 'rubocop:disable Rails' | wc -l
      58
```

I'm not completely sure why it didn't matter in specialist-publisher, for example, but it's possible this check wasn't enabled at the time.

Reply:
I've just double-checked and the equivalent lines in specialist-publisher now also have these warnings.

Reply:
Ah ok, thanks for the explanation. I wonder if that's something worth fixing upstream in https://github.com/alphagov/rubocop-govuk? Out of scope for this PR though 👍